src/index.js

```diff
@@ -1,3 +1,137 @@
+// 전역 변수
+const todoItem = `
+<div class="view">
+  <input class="toggle" type="checkbox"/>
+  <label class="label">새로운 타이틀</label>
+  <button class="destroy"></button>
+</div>
+<input class="edit" value="새로운 타이틀" />
+`;
+let state = {
+    condition: "all",
+    items: [],
+};
+const todoList = document.getElementById("todo-list");
+
+// state객체의 상태에 따라 새로고침 해주는 메소드
+// state.condition 에 따라 각각 다른 리스트를 보여준다.
+// 함수의 마지막에 localStorage에 저장하는 코드가 있음
+function rePaint(state) {
+    todoList.innerHTML = '';
+    let rePaintList = [];
+    if (state.condition === "all") rePaintList = state.items; 
+    if (state.condition === "active") rePaintList = state.items.filter(item=> item.completed === false);
+    if (state.condition === "completed") rePaintList = state.items.filter(item=> item.completed === true);
+
+    rePaintList.forEach((item,i)=>{
+        const classList = [];
+        if (item.completed) classList.push("completed");
+        if (item.editing) classList.push("editing");
+        const li = document.createElement("li");
+        li.innerHTML = todoItem;
+        li.setAttribute("class", classList.join(" "));
+        li.dataset.id = i;
+        li.querySelector(".label").textContent = item.todo;
+        li.querySelector(".edit").value = item.todo;
+        li.querySelector(".edit").addEventListener("keyup", onEdit);
+        li.querySelector(".toggle").addEventListener("change", onChangeCheckBox);
+        if (item.completed) li.querySelector(".toggle").setAttribute("checked", item.completed);
+        li.querySelector(".destroy").addEventListener("click", onClickDestroy);
+
+        li.addEventListener('dblclick',onDoubleClickTodoItem);
+        todoList.append(li);
+    });
```

Comment:
```suggestion
    rePaintList.forEach(({ completed, editing, todo }, i) => {
      const li = document.createElement('li');
      li.outerHTML = `
        <li data-id="${i}" class="${completed ? 'completed' : ''} ${editing ? 'editing' : ''}">
          <div class="view">
            <input class="toggle" type="checkbox" ${completed ? 'checked' : ''} />
            <label class="label">${todo}</label>
            <button class="destroy"></button>
          </div>
          <input class="edit" value="${todo}" />
        </li>
      `;
      li.querySelector(".edit").addEventListener("keyup", onEdit);
      li.querySelector(".toggle").addEventListener("change", onChangeCheckBox);
      li.querySelector(".destroy").addEventListener("click", onClickDestroy);
      li.addEventListener('dblclick',onDoubleClickTodoItem);
      todoList.append(li);
    });
```
이렇게 해보는건 어떨까요? 사실 이벤트 버블링을 이용하면 외부에서 관리할 수 있습니다!
그리고 todoItem이라는 template을 함수로 만들어서 렌더링을 위임할수도 있을 것 같아요~